crates/cli/src/resolver-cli.rs

```diff
@@ -0,0 +1,71 @@
+mod api;
+mod resolver;
+mod utils;
+
+use crate::resolver::{resolve_scarb, TargetType};
+use crate::utils::detect_local_tools;
+
+use camino::Utf8PathBuf;
+use console::{style, Emoji};
+use dialoguer::{theme::ColorfulTheme, Input};
+use dirs::home_dir;
+use std::env;
+
+fn main() -> anyhow::Result<()> {
+    println!(
+        "{} {} Getting project information...",
+        style("[1/4]").bold().dim(),
+        Emoji("📝", "")
+    );
+
+    // Project type + Path entry
+    let target_type = TargetType::ScarbProject; // by default we assume the user is in a scarb project
+    let is_current_dir_scarb = env::current_dir()?.join("scarb.toml").exists();
+    let path = if is_current_dir_scarb {
+        env::current_dir()?.to_str().unwrap().trim().to_string()
+    } else {
+        loop {
+            // TODO, add TargetType::File path input here
+            let input_path = Input::<String>::with_theme(&ColorfulTheme::default())
+                .with_prompt("Enter Path to scarb project root:")
+                .interact_text()
+                .expect("Aborted at path input, terminating...")
+                .trim()
+                .to_string();
+            let mut utf8_path: Utf8PathBuf = Utf8PathBuf::from(&input_path);
+            if utf8_path.starts_with("~") {
+                if let Some(home) = home_dir() {
+                    let home_utf8 = Utf8PathBuf::from_path_buf(home).unwrap();
+                    utf8_path = home_utf8.join(utf8_path.strip_prefix("~").unwrap());
+                }
+            }
+            if utf8_path.exists() {
+                break input_path;
+            } else {
+                println!("Path does not exist. Please try again.");
+            }
+        }
+    };
+    // Resolve path
+    let utf8_path = Utf8PathBuf::from(&path);
+
+    println!(
+        "{} {} Resolving project...",
+        style("[2/4]").bold().dim(),
+        Emoji("🔗", "")
+    );
```

Comment:
Probably want to change the progress indicator to be accurate, so this should be `[2/2]` and the ones before should be `[1/2]`.

Reply:
@cwkang1998 Fixed